### DISCUSSION TITLE
Printer first lay stops at first layer.

### DISCUSSION THREAD

**Discussion by PhenomForgeQX** [sn4k3/UVtools#657] [Q&A] [answered]
Hello,

So I am using UVtools to convert formats and check for issues, however everytime I start a print the printer moves to the first layer exposure position and them stops, the lcd touchscreen becomes unresponsive. 

It basically freezes and the first layer and does not continue the print. I am converting from chitubox to Vlare open format (.osf).

**Comment by sn4k3** [comment]
Which printer are you using?
Send your ctb file as zip

**Reply by PhenomForgeQX** [reply]
Hello,

I am using a Peopoly Phenom Forge with these resin settings 

![image](https://user-images.githubusercontent.com/125074661/218524644-a019dd0d-04bb-49df-8598-920fcdbbb951.png)

Here is a sample part I am trying to print. I slice in Lychee slicer and use UVtools to change to correct resolution and verify settings and convert it to .osf format.
[Feather Clamp.zip](https://github.com/sn4k3/UVtools/files/10724536/Feather.Clamp.zip)

**Reply by sn4k3** [reply]
>  UVtools to change to correct resolution

Is best when resolution is correctly set on slicer, otherwise you may lose quality or have problems.

Can you try prusa slicer with my profiles? It have that printer

**Reply by PhenomForgeQX** [reply]
I created a prusa profile for the Forge and then converted the .sl1 file to .osf in UVtool. UVtools recognized the resolution from Prusaslicer.

However when starting the print it does the same thing, goes to first layer position and then freezes.

**Reply by sn4k3** [reply]
Ok, did you try with official slicer?

**Reply by PhenomForgeQX** [reply]
With the Vlare slicer I have no issues printing when it outputs to .vlr format. It seems .osf is supposed to work with this printer.

**Reply by sn4k3** [reply]
And when exporting to osf on original slicer?

**Reply by PhenomForgeQX** [reply]
Yes, Even when starting with the orignal slicer or some other slicer it produces the same outcome

**Reply by sn4k3** [reply]
Ok, then is not related to UVtools. In that case it must use VLR to work well with your printer.

**Reply by PhenomForgeQX** [reply]
I followed Halpm3Pl3as3's advice down below and .osf files now work with files created from Lychee slicer but not UVtools.

**Reply by sn4k3** [reply]
Ok i will check what is going on, please provide-me two files: 1 osf good from lychee, and same osf but that fails after pass with UVtools

**Reply by Halpm3Pl3as3** [reply]
> Ok i will check what is going on, please provide-me two files: 1 osf good from lychee, and same osf but that fails after pass with UVtools

Hey sn4k3, has the OSF files been provided or are you still waiting?

**Reply by sn4k3** [reply]
not yet...

**Reply by PhenomForgeQX** [reply]
sorry for the delay. Here are the test files, one OFFSET, is osf from lychee slicer that currently works, and another TEST, that is passed through UV tools and does not work.

[slicedtests.zip](https://github.com/sn4k3/UVtools/files/10798799/slicedtests.zip)

**Reply by sn4k3** [reply]
Can't find anything wrong... However, files are from different models. 
Can you provide exactly same file/model, but process original with UVtools and save a copy? Also confirm that the copy doesnt print.

**Comment by Halpm3Pl3as3** [comment]
Glad you posted this. I was pulling my hair trying to figure out what was wrong. I'm experiencing the same issue you've described and thought it was Lychee Slicer.

**Reply by PhenomForgeQX** [reply]
Have you had any luck printing with .osf format files? From what I understand Phenom Forge is supposed to work with them.

**Reply by Halpm3Pl3as3** [reply]
I have had luck with .osf format. I updated the latest firmware https://peopoly.net/pages/support, then I imported Lychee and Vlare profiles (also found on the same page)
Lychee Slicer exports in .osf and for my first few test prints I used Lychee Slicer. I then tried to use the Lychee files in UVTools which is when I started having the issues you described (goes to first layer and does nothing and had to power cycle the printer.)

**Reply by PhenomForgeQX** [reply]
I had no idea Lychee slicer supported OSF. 

Just tried it out and it works great!, thanks for the heads up. Its nice getting out of the Vlare slicer bottleneck.

**Reply by Halpm3Pl3as3** [reply]
I agree! However, Im slowly starting to like Vlare Slicer just because I hate Lychee's pay-to-play with Pro features, paywall. In the two weeks that I've been "forced" to use Vlare, I actually like it since I print only functional parts. 

Sidenote to the Dev(s): This post seems a bit dated but it looks like Vlare posted some Open File Format draft on their site maybe this is of some use to the Dev? https://www.vlare.net/open-format

**Reply by sn4k3** [reply]
The correct draf is here: https://github.com/vlare3d/OSF

**Comment by LoganP1998** [comment]
Hello, I am encountering the same issue using UVTools to detect and repair issues in files for my Phenom Forge.

Vlare and Lychee osf exports print fine straight from the slicer. They will also print fine if they are run through UVTools and are saved straight out with no alterations.
If, however, a repair is done on the file and saved from UVTools, that osf will no longer print and instead freeze on the first layer.

I have attached 3 osf files, all from the same slice from VLare. "Gear_Original" is straight out of Vlare, (Gear_UVTools_Copy) is saved out of UVTools with no alterations, and Gear_Original_Copy_(REPAIRED) is a version where I used the brush to create an error and then detected and repaired it, (this is the one that will not print).


[Gear_UvTest.zip](https://github.com/sn4k3/UVtools/files/11067791/Gear_UvTest.zip)

**Reply by sn4k3** [reply]
I'm not able to see what's wrong. All files open correctly so image should be correct, but something is tripping.
In order to debug this prepare a 1 x 1 x 0.1 mm square and slice it just to one layer on slicer. Then keep original, remove a pixel on the black background with pixel editor and save as copy. 
Do a dry print of UVtools file, if freeze send both files.

**Reply by LoganP1998** [reply]
I've done as instructed. Both the original file and the one with the pixel removed both worked, I also tried a version where the pixel was removed from the square and that started as well. No freezes. Any suggestion on what I should do next to work up to tripping it?

**Reply by sn4k3** [reply]
Try to print this file

[Gear_UVTools_RESAVE.zip](https://github.com/sn4k3/UVtools/files/11070641/Gear_UVTools_RESAVE.zip)

**Reply by LoganP1998** [reply]
That one freezes as well.

**Reply by LoganP1998** [reply]
I tried adding pixels to the debug square both using connected and antialiased brushes and both could print as well. Very strange.

**Reply by sn4k3** [reply]
Try this: [Gear_UVTools_RESAVE.zip](https://github.com/sn4k3/UVtools/files/11070706/Gear_UVTools_RESAVE.zip)

**Reply by LoganP1998** [reply]
That one triggered a freeze!

I have to head off for a couple hours, so I don't have time test this fully at the moment, but I was noticing that some of the Debug Square prints I have been testing by adding pixels with the brush make the printer panel display copies of the brushed section scattered around the LCD. I will test this further later.

**Reply by sn4k3** [reply]
Try: [Gear_UVTools_RESAVE.zip](https://github.com/sn4k3/UVtools/files/11070726/Gear_UVTools_RESAVE.zip)

**Reply by LoganP1998** [reply]
This one prints fine.

**Reply by sn4k3** [reply, marked as answer]
ok, found the problem. Will be on next patch.

**Reply by LoganP1998** [reply]
That's awesome! Thanks so much.

**Reply by LoganP1998** [reply]
Update works perfectly so far!
